### PR TITLE
Migrate Button and ButtonGroup to StyledButton and StyledButtonSet

### PR DESCRIPTION
**Pull request by SudharakaP**
Related to https://github.com/opencollective/opencollective/issues/3223

# Screenshots

<details>
<summary>Screenshots of all Components Migrated</summary>

Screenshot_2020-12-10 generator-jhipster collectives(1)

Screenshot_2020-12-10 Agora Collective - Open Collective(2)

Screenshot_2020-12-10 generator-jhipster collectives

Screenshot_2020-12-10 Open Collective Hosts - Open Collective

Screenshot_2020-12-10 Gift Card Redeemed - Open Collective

Screenshot_2020-12-10 Agora Collective - Open Collective(1)

Screen Shot 2020-12-10 at 10 40 07

Screenshot_2020-12-10 Create a New Agora Collective Event - Open Collective

Screenshot_2020-12-10 Agora Collective - Open Collective
</details>

### CODE REVIEW

**Comment by SudharakaP** [comment]
@Betree : Just to note that I've also included two Form element removals to this PR as they seem simple enough. 😄

File: components/edit-collective/sections/CollectiveGoals.js [hunk 1]
```diff
@@ -167,55 +166,53 @@ class CollectiveGoals extends React.Component {
 
     return (
       <Container mt={4} pb={4} borderBottom={BORDER} key={`goal-${index}-${goal.key}`}>
-        <Form>
```
Comment (Betree) [inline components/edit-collective/sections/CollectiveGoals.js:170]: Any chance we can replace with a simple `form`?

File: components/edit-collective/sections/Host.js [hunk 2]
```diff
@@ -132,12 +131,12 @@ class Host extends React.Component {
           {collective.stats.balance === 0 && (
             <Fragment>
               <p>
-                <Button bsStyle="primary" type="submit" onClick={() => this.changeHost()} className="removeHostBtn">
```
Comment (Betree) [inline components/edit-collective/sections/Host.js:135]: It's good to keep the `type="submit"` if already set, makes things explicit in the DOM

File: components/edit-collective/sections/Host.js [hunk 3]
```diff
@@ -172,17 +171,16 @@ class Host extends React.Component {
                     />
                   </p>
                   <p>
-                    <Button
-                      bsStyle="primary"
-                      type="submit"
```
Comment (Betree) [inline components/edit-collective/sections/Host.js:177]: It's good to keep the `type="submit"` if already set, makes things explicit in the DOM

File: components/edit-collective/sections/Host.js [hunk 4]
```diff
@@ -226,14 +224,13 @@ class Host extends React.Component {
                   {collective.stats.balance === 0 && (
                     <Fragment>
                       <p>
-                        <Button
-                          bsStyle="primary"
-                          type="submit"
```
Comment (Betree) [inline components/edit-collective/sections/Host.js:231]: It's good to keep the `type="submit"` if already set, makes things explicit in the DOM

File: components/edit-collective/sections/Host.js [hunk 5]
```diff
@@ -363,12 +360,12 @@ class Host extends React.Component {
               {selectedOption === 'selfHost' && LoggedInUser && (
                 <Flex justifyContent="space-between" alignItems="flex-end" mt={3}>
                   <Box>
-                    <Button bsStyle="primary" type="submit" onClick={() => this.changeHost({ id: collective.id })}>
```
Comment (Betree) [inline components/edit-collective/sections/Host.js:366]: It's good to keep the `type="submit"` if already set, makes things explicit in the DOM

**Review by Betree — COMMENTED** [review]
Looking good! Some comments below, mainly to improve accessibility

File: components/MembersWithData.js [hunk 6]
```diff
@@ -137,36 +142,13 @@ class MembersWithData extends React.Component {
...
-                bsStyle={this.state.role === 'BACKER' ? 'primary' : 'default'}
-                onClick={() => this.refetch('BACKER')}
-              >
-                <FormattedMessage id="members.paid" defaultMessage="backers" />
-              </Button>
-            </ButtonGroup>
+            <StyledButtonSet
+              className="filterBtnGroup"
+              items={['all', 'admin', 'members', 'paid']}
+              onChange={() => (!this.state.role ? this.refetch() : this.refetch(this.state.role))}
```
[the remark this answers is not in the document]
Comment (SudharakaP) [reply components/MembersWithData.js:148]: @Betree : I am doubtful if this filtering system is shown at all; I don't see anywhere where we pass a value to the `refetch` prop nor a situation where both `tier` and `role` is undefined. 🤔 When you go to the `banner.html` is there a way you could view this filter? 🤔

[the remark this answers is not in the document]
Comment (Betree) [reply components/MembersWithData.js:125]: You're right, it seems that we removed all the other instances where we used `MembersWithData`. Ok to get rid of these then :+1:

[the remark this answers is not in the document]
Comment (SudharakaP) [reply components/MembersWithData.js]: Thanks for the confirmation. It's done 😄

File: components/edit-collective/CreateHostForm.js [hunk 7]
```diff
@@ -139,23 +139,22 @@ class CreateHostForm extends React.Component {
         {!host && (
           <Fragment>
             <CreateOrganizationForm onChange={org => this.handleChange('organization', org)} />
-            <Button
-              bsStyle="primary"
-              type="submit"
+            <StyledButton
+              buttonStyle="primary"
```
Comment (Betree) [inline components/edit-collective/CreateHostForm.js:143]: We shoud keep this

```suggestion
              buttonStyle="primary"
              type="submit"
```

**Review by Betree — COMMENTED** [review]
Just a small one to change, everything else is good to merge :sun_with_face:

File: components/edit-collective/CreateHostForm.js (hunk 7 above)
Comment (SudharakaP) [reply components/edit-collective/CreateHostForm.js:143]: @Betree : Done. Thanks for pointing it out. 👍🏽

**Review by Betree — APPROVED** [review]
Nice job :+1:

**Comment by SudharakaP** [comment]
> Nice job +1

Thanks. 👍🏽

**Pull request merged**